common/lib/xmodule/xmodule/x_module.py

```diff
@@ -398,7 +398,7 @@ def get_content_titles(self):
         the display_names of all its leaf nodes.
         """
         if self.has_children:
-            return sum((child.get_content_titles() for child in self.get_children()), [])
+            return sum((child.get_content_titles() for child in self.get_display_items()), [])
```

Comment:
from a quick look at the code i think `get_display_items()` is more correct than what was there before (has the tooltip stuff been working correctly for ABTestModules til now?)

However, `get_display_items` doesn't solve the access check condition that needs to be added; I believe a `has_access` call needs to be done somewhere (similar to how it's done where we generate TOC) because the x_module methods don't know anything about the present user, and shouldn't.

unfortunately this has_access call may be expensive to make, but i don't see a way around it.